nemo/collections/common/data/lhotse/dataloader.py

```diff
@@ -186,16 +192,13 @@ def get_lhotse_dataloader_from_config(
     # Expands cuts if multiple translations are provided.
     cuts = CutSet(LazyFlattener(cuts.map(_flatten_alt_text, apply_fn=None)))
 
-    if config.use_multimodal_sampling:
-        assert (
-            tokenizer is not None
-        ), "You must pass a tokenizer to `get_lhotse_dataloader_from_config` in order to read text-only datasets (enabled via use_multimodal_dataloading)"
+    if tokenizer is not None:
         from nemo.collections.asr.data.audio_to_text_lhotse import TokenizerWrapper
 
         if not isinstance(tokenizer, TokenizerWrapper):
             tokenizer = TokenizerWrapper(tokenizer)
-        # Note this code can also pre-tokenize the text in cuts, but for now we disable it with apply_fn.
-        cuts = cuts.map(partial(tokenize, tokenizer=tokenizer), apply_fn=is_text)
+        cuts = cuts.map(partial(tokenize, tokenizer=tokenizer), apply_fn=None)
+        cuts = cuts.filter(TokenPerSecondFilter(config.min_tps, config.max_tps))
```

Comment:
TokenPerSecondFilter needs to have the same min_tps and max_tps parameters as https://github.com/NVIDIA/NeMo/pull/9763/files#diff-7cf5386bc408c1b18dc8c78908e65d75e01aff39680de8fca6b37ac1350add4eR294 for expected performance, right? Assuming that you use estimate_durations_bins_2d.py to create your bins (which I don't see why you wouldn't).

Reply:
Yeah, if you use oomptimizer but don't specify max_tps, and you e.g. change the training dataset to one that has longer utterances, you'll likely run into OOM. Both estimate_duration_bins[_2d] and oomptimizer print out suggested `max_tps`.